embedly/embedly/extract.py

```diff
@@ -23,7 +27,12 @@ def _get_cached_url(self, url):
         cached_data = self.redis_client.get(cache_key)
 
         if cached_data is not None:
-            return json.loads(cached_data)
+            try:
+                return json.loads(cached_data)
+            except ValueError:
+                raise URLExtractorException(
```

Comment:
I think that if there is a partial failure, we should return a partial result.
There is the situation where
In which case, the client should know that some urls weren't return given the returned data schema.